ddtrace/_trace/span.py

```diff
@@ -631,11 +634,28 @@ def set_link(
             )
         )
 
-    def _set_link_object(self, link: SpanLink) -> None:
-        # We will be changing this behavior to allow certain kinds of span
-        # links to coexist in the _links list even if they have the same
-        # span_id. For now, we are basically reimplementing the old
-        # dictionary-like behavior.
+    def _add_span_pointer(
+        self,
+        pointer_kind: str,
+        pointer_direction: _SpanPointerDirection,
+        pointer_hash: str,
+        extra_attributes: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        # This is a Private API for now.
+
+        self._set_link_or_append_pointer(
+            _SpanPointer(
+                pointer_kind=pointer_kind,
+                pointer_direction=pointer_direction,
+                pointer_hash=pointer_hash,
+                extra_attributes=extra_attributes,
+            )
+        )
+
+    def _set_link_or_append_pointer(self, link: Union[SpanLink, _SpanPointer]) -> None:
```

Comment:
Do we need a union here? _SpanPointers should also be a SpanLink
```suggestion
    def _set_link_or_append_pointer(self, link: SpanLink) -> None:
```

Reply:
yup, but i wanted to extra clear that we can store either a span link or a span pointer here. currently span pointers are implemented as span links, but i'd argue that's an implementation detail.

Reply:
i've actually added a similar typehint to the `_links` attribute. please let me know if you would prefer i went the other way.

Reply:
Doesn't really matter to me. We can keep it.